src/dex/fluid-dex/fluid-dex.ts

```diff
@@ -0,0 +1,881 @@
+    readonly network: Network,
+    readonly dexKey: string,
+    readonly dexHelper: IDexHelper,
+  ) {
+    super(dexHelper, dexKey);
+    this.logger = dexHelper.getLogger(dexKey);
+    this.FluidCommonAddresses = new FluidDexCommonAddresses(
+      'FluidDex',
+      FluidDexConfig['FluidDex'][network].commonAddresses,
+      network,
+      dexHelper,
+      this.logger,
+    );
+    this.pools = FluidDexConfig['FluidDex'][network].pools;
+    this.adapters = Adapters[network] || {};
+    this.iFluidDexPool = new Interface(FluidDexPoolABI);
+  }
+
+  async fetchFluidDexPools(blockNumber: number): Promise<FluidDexPool[]> {
+    const poolsFromResolver =
+      await this.FluidCommonAddresses.getStateOrGenerate(blockNumber, false);
+
+    return poolsFromResolver.map(pool => ({
+      id: `FluidDex_${pool.address}`,
+      address: pool.address,
+      token0: pool.token0,
+      token1: pool.token1,
+    }));
+  }
+
+  async updatePoolAndEventPool(blockNumber: number) {
+    this.pools = await this.fetchFluidDexPools(blockNumber);
+    for (const pool of this.pools) {
+      if (!this.eventPools[pool.id]) {
+        this.eventPools[pool.id] = new FluidDexEventPool(
+          'FluidDex',
+          pool.address,
+          this.FluidCommonAddresses.commonAddresses,
+          this.network,
+          this.dexHelper,
+          this.logger,
+        );
+      }
+    }
+  }
+
+  // Initialize pricing is called once in the start of
+  // pricing service. It is intended to setup the integration
+  // for pricing requests. It is optional for a DEX to
+  // implement this function
+  async initializePricing(blockNumber: number) {
+    Object.entries(this.eventPools).forEach(([id, eventPool]) => {
+      eventPool.getStateOrGenerate(blockNumber, false);
+    });
+    await this.updatePoolAndEventPool(blockNumber);
+  }
+
+  // Returns the list of contract adapters (name and index)
+  // for a buy/sell. Return null if there are no adapters.
+  getAdapters(side: SwapSide): { name: string; index: number }[] | null {
+    return this.adapters[side] ? this.adapters[side] : null;
+  }
+
+  decodePools = (result: MultiResult<BytesLike> | BytesLike): Pool[] => {
+    return generalDecoder(
+      result,
+      ['tuple(address pool, address token0, address token1)[]'],
+      undefined,
+      decoded => {
+        return decoded.map((decodedPool: any) => ({
+          address: decodedPool[0][0].toLowerCase(),
+          token0: decodedPool[0][1].toLowerCase(),
+          token1: decodedPool[0][2].toLowerCase(),
+        }));
+      },
+    );
+  };
+
+  // Returns list of pool identifiers that can be used
+  // for a given swap. poolIdentifiers must be unique
+  // across DEXes. It is recommended to use
+  // ${dexKey}_${poolAddress} as a poolIdentifier
+  async getPoolIdentifiers(
+    srcToken: Token,
+    destToken: Token,
+    side: SwapSide,
+    blockNumber: number,
+  ): Promise<string[]> {
+    await this.updatePoolAndEventPool(blockNumber);
```

Comment:
should this search in existing pools instead? can we just trigger a fetch from resolver here?